secure.go

```diff
@@ -290,6 +295,10 @@ func (s *Secure) processRequest(w http.ResponseWriter, r *http.Request) (http.He
 	}
 
 	// Allowed hosts check.
+	if s.opt.AllowedHostsFunc != nil && !s.opt.IsDevelopment {
+		s.opt.AllowedHosts = append(s.opt.AllowedHosts, s.opt.AllowedHostsFunc()...)
+	}
+
```

Comment:
```suggestion
    combinedAllowedHosts = s.opt.AllowedHosts
	if s.opt.AllowedHostsFunc != nil {
		combinedAllowedHosts = append(combinedAllowedHosts, s.opt.AllowedHostsFunc()...)
	}
```

This function is called on every request, so I think the current implementation will just append the function results every time. I'm thinking we can create a temporary variable and combine the static and dynamic lists instead.

Reply:
Hm, I actually need it to be called on every request. In my case, the list changes.

Reply:
My suggested change will still call the function on every request. The way the code is written now will append the results of the function call into `s.opt.AllowedHosts` on every request... so that array will grow and grow and grow! Instead of appending to `s.opt.AllowedHosts`, we can create a new variable that combines both the static list and the function call results. Then on lines 302 and 312 we can update the references from `s.opt.AllowedHosts` to the new variable we created (I called it `combinedAllowedHosts` in my suggestion above)

Reply:
Oh, of course, I see. Looked into the source more closely, thanks

Reply:
Am actually wondering if we can modify this to pass in the request to the function, what do you think? Or have another function` isOriginAllowed(host string) `

Reply:
I like the idea of passing the request in! I'll try to prototype that soon and see how it fits in.